src/shogun/classifier/Perceptron.cpp

```diff
@@ -44,28 +36,22 @@ CPerceptron::~CPerceptron()
 {
 }
 
-bool CPerceptron::train_machine(CFeatures* data)
+void CPerceptron::train_machine(CFeatures* features, CLabels* labels)
 {
-	ASSERT(m_labels)
+	if (!features->has_property(FP_DOT))
```

Comment:
why not `as`? would be a bit cleaner. Training is expensive, so the additional costs shouldnt matter or?

Reply:
we can provide more informative error message here, but `as` is as good